src/main/java/org/takes/misc/VerboseList.java

```diff
@@ -0,0 +1,205 @@
+    }
+
+    @Override
+    public boolean remove(final Object obj) {
+        return this.origin.remove(obj);
+    }
+
+    @Override
+    public boolean containsAll(final Collection<?> coll) {
+        return this.origin.containsAll(coll);
+    }
+
+    @Override
+    public boolean addAll(final Collection<? extends T> coll) {
+        return this.origin.addAll(coll);
+    }
+
+    @Override
+    public boolean addAll(final int index, final Collection<? extends T> coll) {
+        try {
+            return this.origin.addAll(index, coll);
+        } catch (final IndexOutOfBoundsException ex) {
+            throw new IllegalArgumentException(this.message, ex);
+        }
+    }
+
+    @Override
+    public boolean removeAll(final Collection<?> coll) {
+        return this.origin.removeAll(coll);
+    }
+
+    @Override
+    public boolean retainAll(final Collection<?> coll) {
+        return this.origin.retainAll(coll);
+    }
+
+    @Override
+    public void clear() {
+        this.origin.clear();
+    }
+
+    @Override
+    public T get(final int index) {
+        try {
+            return this.origin.get(index);
+        } catch (final IndexOutOfBoundsException ex) {
+            throw new IllegalArgumentException(this.message, ex);
+        }
+    }
+
+    @Override
+    public T set(final int index, final T element) {
+        try {
+            return this.origin.set(index, element);
+        } catch (final IndexOutOfBoundsException ex) {
+            throw new IllegalArgumentException(this.message, ex);
+        }
+    }
+
+    @Override
+    public void add(final int index, final T element) {
+        try {
+            this.origin.add(index, element);
+        } catch (final IndexOutOfBoundsException ex) {
+            throw new IllegalArgumentException(this.message, ex);
+        }
+    }
+
+    @Override
+    public T remove(final int index) {
+        try {
+            return this.origin.remove(index);
+        } catch (final IndexOutOfBoundsException ex) {
+            throw new IllegalArgumentException(this.message, ex);
+        }
+    }
+
+    @Override
+    public int indexOf(final Object obj) {
+        return this.origin.indexOf(obj);
+    }
+
+    @Override
+    public int lastIndexOf(final Object obj) {
+        return this.origin.lastIndexOf(obj);
+    }
+
+    @Override
+    public ListIterator<T> listIterator() {
+        return this.origin.listIterator();
+    }
+
+    @Override
+    public ListIterator<T> listIterator(final int index) {
+        try {
+            return this.origin.listIterator(index);
+        } catch (final IndexOutOfBoundsException ex) {
+            throw new IllegalArgumentException(this.message, ex);
+        }
+    }
+
+    @Override
+    public List<T> subList(final int fridx, final int toidx) {
+        try {
+            return this.origin.subList(fridx, toidx);
+        } catch (final IndexOutOfBoundsException ex) {
+            throw new IllegalArgumentException(this.message, ex);
```

Comment:
@Happy-Neko Why wrap this exception instead of rethrowing it? `IndexOutOfBoundException` is much more specific.